src/Driver/Mysqli/Result.php

```diff
@@ -40,11 +45,17 @@ final class Result implements ResultInterface
     /**
      * @internal The result can be only instantiated by its driver connection or statement.
      *
+     * @param Statement|null $statementReference This reference is to not trigger destruction
+     * of the Statement object too early.
+     *
      * @throws Exception
      */
-    public function __construct(mysqli_stmt $statement)
-    {
-        $this->statement = $statement;
+    public function __construct(
+        mysqli_stmt $statement,
+        ?Statement $statementReference = null,
```

Comment:
```suggestion
        ?Statement $statementReference = null
```

PHP 7 does not support trailing commas in method signatures.